frontend/src/components/BookTable.tsx

```diff
@@ -0,0 +1,624 @@
+    noRowsLabel: t('No rows'),
+    noResultsOverlayLabel: t('No results found.'),
+    errorOverlayDefaultLabel: t('An error occurred.'),
+    toolbarColumns: t('Columns'),
+    toolbarColumnsLabel: t('Select columns'),
+    columnsPanelTextFieldLabel: t('Find column'),
+    columnsPanelTextFieldPlaceholder: t('Column title'),
+    columnsPanelDragIconLabel: t('Reorder column'),
+    columnsPanelShowAllButton: t('Show all'),
+    columnsPanelHideAllButton: t('Hide all'),
+    filterPanelAddFilter: t('Add filter'),
+    filterPanelDeleteIconLabel: t('Delete'),
+    filterPanelLinkOperator: t('Logic operator'),
+    filterPanelOperators: t('Operator'),
+    filterPanelOperatorAnd: t('And'),
+    filterPanelOperatorOr: t('Or'),
+    filterPanelColumns: t('Columns'),
+    filterPanelInputLabel: t('Value'),
+    filterPanelInputPlaceholder: t('Filter value'),
+    filterOperatorContains: t('contains'),
+    filterOperatorEquals: t('equals'),
+    filterOperatorStartsWith: t('starts with'),
+    filterOperatorEndsWith: t('ends with'),
+    filterOperatorIs: t('is'),
+    filterOperatorNot: t('is not'),
+    filterOperatorAfter: t('is after'),
+    filterOperatorOnOrAfter: t('is on or after'),
+    filterOperatorBefore: t('is before'),
+    filterOperatorOnOrBefore: t('is on or before'),
+    filterOperatorIsEmpty: t('is empty'),
+    filterOperatorIsNotEmpty: t('is not empty'),
+    filterOperatorIsAnyOf: t('is any of'),
+    filterValueAny: t('any'),
+    filterValueTrue: t('true'),
+    filterValueFalse: t('false'),
+    columnMenuLabel: t('Menu'),
+    columnMenuShowColumns: t('Show columns'),
+    columnMenuFilter: t('Filter'),
+    columnMenuHideColumn: t('Hide'),
+    columnMenuUnsort: t('Unsort'),
+    columnMenuSortAsc: t('Sort by ASC'),
+    columnMenuSortDesc: t('Sort by DESC'),
+    columnHeaderFiltersLabel: t('Show filters'),
+    columnHeaderSortIconLabel: t('Sort'),
+    booleanCellTrueLabel: t('yes'),
+    booleanCellFalseLabel: t('no'),
+  };
+}
+
+interface Props {
+  loading: boolean;
+  orders: Order[];
+  type: number;
+  currency: number;
+  maxWidth: number;
+  maxHeight: number;
+}
+
+const BookTable = ({
+  loading,
+  orders,
+  type,
+  currency,
+  maxWidth,
+  maxHeight,
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const history = useHistory();
+
+  const fontSize = theme.typography.fontSize;
+
+  // all sizes in 'em'
+  const verticalHeightFrame = 6.9075;
+  const verticalHeightRow = 3.25;
+  const defaultPageSize = Math.max(
+    Math.floor((maxHeight - verticalHeightFrame) / verticalHeightRow),
+    1,
+  );
+  const height = defaultPageSize * verticalHeightRow + verticalHeightFrame;
+
+  const [pageSize, setPageSize] = useState(0);
+  const [useDefaultPageSize, setUseDefaultPageSize] = useState(true);
+  useEffect(() => {
+    if (useDefaultPageSize) {
+      setPageSize(defaultPageSize);
+    }
+  });
```

Comment:
}, []);

Reply:
For some reason this change makes the `pageSize` stay as 0.